src/main.rs

```diff
@@ -23,7 +23,7 @@ use std::{
     str::FromStr,
 };
 use structopt::clap::arg_enum;
-use structopt::{clap::AppSettings, StructOpt};
+use structopt::{StructOpt, clap::AppSettings};
```

Comment:
suggestion: eliminate the `structopt` dependency and use `clap_derive`.

`structopt` is now in maintenance mode. `clap_derive` is (almost) a drop-in replacement and will get any bug fixes, new features, etc.